volatility3/framework/plugins/windows/processghosting.py

```diff
@@ -33,52 +35,163 @@ def get_requirements(cls):
+        Tuple[int, Optional[int], Optional[int], int, Optional[str]], None, None
+    ]:
+        """
+        Returns process or vad info for ghosting files
+
+        Args:
+            proc:
+            mapped_files: A dictionary mapping vad base addreses to the path and vad instance for the process
+
+        Return:
+            A Generator of tuples of the file object address, the delete pending state, delete on close state, base address of the VAD, and the path
+        """
+        # check the direct file object of the process
+        yield from cls._process_checks(proc, mapped_files)
+
+        # walk each vad, check if it is pending delete or has its delete on close bit set
+        for vad_base, (path, vad) in mapped_files.items():
+            # these checks have no meaning for private memory areas
+            if vad.get_private_memory() == 1:
+                continue
+
+            try:
+                if vad.has_member("ControlArea"):
+                    control_area = vad.ControlArea
+                elif vad.has_member("Subsection"):
+                    control_area = vad.Subsection.ControlArea
+                # We got here from a short vad, likely smear
+                else:
+                    continue
+            except exceptions.InvalidAddressException:
                 vollog.debug(
-                    f"Invalid delete_pending value {delete_pending} found for {process_name} {proc.UniqueProcessId}"
+                    f"Unable to get control area for vad at base {vad_base:#x} for process with pid {proc.UniqueProcessId}"
                 )
+                continue
+
+            for file_object_address, delete_pending, delete_on_close in cls._vad_checks(
+                control_area, path
+            ):
+                yield format_hints.Hex(
+                    file_object_address
+                ), delete_pending, delete_on_close, vad_base
+
+    def _generator(self, procs):
+        kernel = self.context.modules[self.config["kernel"]]
 
-            # delete_pending besides 0 or 1 = smear
-            if file_object == 0 or delete_pending == 1:
-                path = renderers.UnreadableValue()
-                if file_object:
-                    with contextlib.suppress(exceptions.InvalidAddressException):
-                        path = file_object.FileName.String
-
-                yield (
-                    0,
-                    (
-                        proc.UniqueProcessId,
-                        process_name,
-                        format_hints.Hex(file_object),
-                        delete_pending,
-                        path,
-                    ),
+        has_imagefilepointer = kernel.get_type("_EPROCESS").has_member(
+            "ImageFilePointer"
+        )
+        if not has_imagefilepointer:
+            vollog.warning(
+                "ImageFilePointer checks are only supported on Windows 10 builds when the ImageFilePointer member of _EPROCESS is present"
```

Comment:
Is this specific to windows 10, or does it apply to 7 or 11?  Seems strange to be saying windows 10 specifically here?  Can this error never happen on a different version of windows?

Reply:
Nice catch. Should be "10+" like other plugins.

Reply:
addressed in latest commit.